lib/write-to-stream-test.ts

```diff
@@ -60,4 +60,22 @@ describe('writeToStream', () => {
     stream.resume()
     await writeToStream(stream, values)
   })
+  it('rejects on stream errors', async () => {
+    const values = [1, 2, 3, 4]
+    const stream = new Writable({
+      highWaterMark: 0,
+      objectMode: true,
+      write(chunk, encoding, cb) {
+        setImmediate(() => cb(new Error('Stream error')))
+      },
+    })
+
+    try {
+      await writeToStream(stream, values)
+    } catch (e) {
+      assert.equal(e.message, 'Stream error')
+      return
+    }
+    throw new Error('No error caught')
+  })
```

Comment:
I think that you can probably grab the tests from my PR #33 

I think that the tests about leaving no dangling listeners and reading as little as possible from the iterator would be very valuable